pkg/prometheus/server/thanos_sidecar_config.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/ptr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+	prompkg "github.com/prometheus-operator/prometheus-operator/pkg/prometheus"
+)
+
+const (
+	ThanosConfigDir                                  = "/etc/thanos/config"
+	ThanosPrometheusHTTPClientConfigFileName         = "prometheus.http-client-file.yaml"
+	ThanosPrometheusHTTPClientConfigSecretNameSuffix = "thanos-prometheus-http-client-file"
```

Comment:
(nit) no need to make the constants public.
```suggestion
	thanosConfigDir                                  = "/etc/thanos/config"
	thanosPrometheusHTTPClientConfigFileName         = "prometheus.http-client-file.yaml"
	thanosPrometheusHTTPClientConfigSecretNameSuffix = "thanos-prometheus-http-client-file"
```